tests/testpeeridauth.nim

```diff
@@ -0,0 +1,57 @@
+{.used.}
+
+# Nim-Libp2p
+# Copyright (c) 2023 Status Research & Development GmbH
+# Licensed under either of
+#  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+#  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+# at your option.
+# This file may not be copied, modified, or distributed except according to
+# those terms.
+
+{.push raises: [].}
+
+import json, uri
+import chronos
+import chronos/apps/http/httpclient
+import ../libp2p/[stream/connection, upgrademngrs/upgrade, peeridauth, wire]
+
+import ./helpers
+
+const
+  AuthPeerURL = "https://registration.libp2p.direct/v1/_acme-challenge"
+  HttpPeerAuthFailed = 401
+
+suite "PeerID Auth":
+  var api {.threadvar.}: PeerIDAuthApi
+  var peerInfo {.threadvar.}: PeerInfo
+
+  asyncTeardown:
+    await api.close()
+    checkTrackers()
+
+  asyncSetup:
+    let rng = newRng()
+    api = PeerIDAuthApi.new(rng)
+    peerInfo = PeerInfo.new(PrivateKey.random(PKScheme.RSA, rng[]).get())
+
+  asyncTest "test peerID send":
+    let payload =
+      %*{
+        "identifiers": [
+          {
+            "type": "dns",
+            "value":
+              "*.k51qzi5uqu5dj8c5nhiw2oceam0uebustsj7s36kjxwtscngp0y126o3b95mh9.libp2p.direct",
+          }
+        ]
+      }
+
+    let (bearer, responseWithoutBearer) =
+      await api.send(parseUri(AuthPeerURL), peerInfo, payload)
+    check responseWithoutBearer.status != HttpPeerAuthFailed
+    doAssert bearer.token.len > 0
+
+    let (_, responseWithBearer) =
+      await api.send(parseUri(AuthPeerURL), peerInfo, payload, bearer)
+    check responseWithBearer.status != HttpPeerAuthFailed
```

Comment:
why is user forced to use `responseWithBearer.status` to check if send, unfortunately named, is successful? this is causing user to process errors in two ways: 1st) one that he must - exceptions 2nd) checking `responseWithBearer.status`
can send throw exception in case when action is successful?